### DISCUSSION TITLE
Create seeders for whole DB

### DISCUSSION THREAD

**Discussion by anegoda1995** [RetroAchievements/RAWeb#853] [Ideas]
I want to contribute in project but when I set up by docker-compose, DB is empty. It is very hard to start to work on some bug or feature, because you can't perform some actions like on production.

After docker-compose up
![image](https://user-images.githubusercontent.com/10788396/142335133-6df107e6-4625-462d-8ca5-43eef09d342d.png)

Production:
![image](https://user-images.githubusercontent.com/10788396/142335433-aaa1b171-4efe-4656-89df-2f45a1de991b.png)

I started to explore codebase, DB and see that I need some data to work with.
My idea is to create seeders that will seed the DB by random faker data every docker-compose up. It will open doors to test whole system.

My proposition is to use faker library that allow to generate thousands of data depends on needs (emails, cities, names, phones etc)
https://github.com/fzaninotto/Faker

**Comment by Jamiras** [comment]
There used to be a sanitized DB dump from 2018 to seed local environments. Then there's a series of sql files in the database subdirectory to upgrade it to current state. I don't know if that file is still available anywhere, or if there's a newer one, or an alternate preferred way.

**Reply by anegoda1995** [reply]
> There used to be a sanitized DB dump from 2018 to seed local environments. Then there's a series of sql files in the database subdirectory to upgrade it to current state. I don't know if that file is still available anywhere, or if there's a newer one, or an alternate preferred way.

You right, after MySQL container will be initialized all SQL commands from **database** folder will be executed, because docker-compose.yml have this string:
`- ./database:/docker-entrypoint-initdb.d/`
 and first file that will be executed it is database/20180212_170235_RA_Template.sql that will create all tables in DB, after this file, will be executed some migrations that will modify DB for consistent with _current production state_.

It is all about SCHEMA, but not about DATA. I want to create seeders for seed DB by dummy data.

**Reply by Jamiras** [reply]
Yes, I understand you want data. We had a sanitized DB dump from 2018. My dev site is populated with it, but I don't have the source file any more, and don't remember where I got it. I guess I could dump my DB to provide something for your use, but I'm not sure what the official process should be.

**Reply by anegoda1995** [reply]
Please dump your DB with populated data that I can use.

But whatever it is better to create **seeders** that will populate data with every docker-compose re-up. It is more testable. Moreover seeders can be modified when DB structure will be changed it is more convinient.

**Reply by Tsearo** [reply]
Here is the sanitized dump we use for populating data
**

and a separate one for populating set requests
**

**Reply by anegoda1995** [reply]
> Here is the sanitized dump we use for populating data
> **
> 
> and a separate one for populating set requests 
> **

Thank you. I tried it and it have all needed data.
First time my dump was imported in 2 hours with default docker mysql container settings, but I did some modifications and now it is importing in ~14 minutes
Maybe useful for someone (for 16 RAM and 16 threads):
1. Modify your mysql.cnf like this
```
[mysqld]
skip-host-cache
skip-name-resolve
innodb_buffer_pool_size = 2G
innodb_log_buffer_size = 512M
innodb_log_file_size = 1G
innodb_write_io_threads = 8
innodb_flush_log_at_trx_commit = 0
```
2. Do command **docker-compose up -d --build --force-recreate**

**Comment by anegoda1995** [comment]
Thank you for replies and help with dump, but anyway, I think even 14 minutes to import it is very much for development with docker. I started to create seeders that will fill up the DB with fake data and it will seed the DB in less than 1 minute after every docker-compose. It will allow us to fix bugs/add features quickest and help new contributors to start contribute in project in several minutes.
